Featured accessory

fnoah opened the pull request:
#2 and #7

BalestraPatrick commented on Sources/App/Controllers/HomeController.swift:
```diff
@@ -0,0 +1,52 @@
+        accessories = try Accessory.makeQuery().filter("approved", true).limit(visibleAccessoriesLimit).all().sorted(by: { $0.date.compare($1.date) == .orderedDescending })
+
+        // Get featured accessory from item id
+        let featuredAccessoryDbId = 26
```
You can easily name this property `featuredAccessoryId` so we don't expose the fact it's stored in the database

BalestraPatrick commented on Sources/App/Controllers/HomeController.swift:
```diff
@@ -0,0 +1,52 @@
+        let featuredAccessoryImage: String
+        
+        // Limit of fetched items for accessories query
+        let visibleAccessoriesLimit = 20
```
On a second thought, since the number of accessories in each row is 3, let's set the limit to 18 so we have exactly 6 rows.

BalestraPatrick commented on Sources/App/Controllers/HomeController.swift:
```diff
@@ -0,0 +1,52 @@
+        let visibleAccessoriesLimit = 20
+
+        // Get limited amout of accessoires sorted by date
+        accessories = try Accessory.makeQuery().filter("approved", true).limit(visibleAccessoriesLimit).all().sorted(by: { $0.date.compare($1.date) == .orderedDescending })
```
Very nice use of the limit function 👍

BalestraPatrick commented on Resources/Views/home.leaf:
```diff
@@ -0,0 +1,87 @@
+            </div>
+
+            <!-- Content -->
+            <div class="col-md-9">
```
Can we wrap this featured item div in a `#if()` to check if the featured item exists? In this way, in case we want to temporarily remove the accessory, we can just remove the featured product from the `Node` which is rendered and avoid to modify this `leaf` file every time 😄

BalestraPatrick commented:
Looks good @fnoah! Do you have any idea why the tests are failing? Seems like `Droplet.testable()` is crashing when initializing the `Droplet` but you don't seem to have changed anything there 🤔

fnoah commented:
I just added the time ago string to each accessory. 
The unit tests in Xcode all run without any problems. I suppose the failing tests have something to do with the new controller I added. But unfortunately, I have no idea why exactly the tests are failing 😕

BalestraPatrick commented on Sources/App/Extensions/Date+Utils.swift:
```diff
@@ -0,0 +1,47 @@
+
+import Vapor
+
+extension Date {
```
This seems to work fine, but maybe we can leverage the already provided APIs in `NSCalendar` and `NSDate`. Take a look at this function, it seems to do exactly what you need: https://gist.github.com/minorbug/468790060810e0d29545

BalestraPatrick commented:
Great work on fixing the tests, I totally missed that myself 🥇

fnoah commented:
Thank you @BalestraPatrick, I absolutely forgot to check if I can solve the problem using default APIs.

BalestraPatrick commented:
@fnoah Looks good! 
If you can just remove the hardcoded featured accessory for now (comment it out or something like that) so that as soon as we have a real product to feature (still working on getting the first partnership), we can bring it back. After that, we can merge it 🚀

fnoah commented:
@BalestraPatrick Okay, I commented out the hardcoded featured accessory

BalestraPatrick commented:
Awesome, it's deploying right now! 😎
Thank you very much for your contribution @fnoah!  💯
